apps/arweave/src/ar_tx_queue.erl

```diff
@@ -0,0 +1,284 @@
+					_ ->
+						%% Send transactions to the "max_propagation_peers" best peers,
+						%% ?TX_PROPAGATION_PARALLELIZATION peers at a time. The resulting
+						%% maximum for outbound connections is ?TX_PROPAGATION_PARALLELIZATION * num_emitters.
+						lists:foreach(
+							fun(_) ->
+								erlang:send(?MODULE, {emit_tx_to_peer, TX})
+							end,
+							lists:seq(1, ?TX_PROPAGATION_PARALLELIZATION)
+						),
+						TXID = TX#tx.id,
+						ar:info(
+							[
+								{sending_tx_to_external_peers, ar_util:encode(TXID)},
+								{peers, length(Peers)}
+							]
+						),
+						NewEmitMap = EmitMap#{ TXID => #{ peers => Peers, started_at => erlang:timestamp() } },
+						State#state { tx_queue = NewQ, size = Size - TXSize, emit_map = NewEmitMap }
+				end
+		end,
+	{noreply, NewState};
+
+handle_info({emit_tx_to_peer, TX}, State = #state { emit_map = EmitMap }) ->
+	TXID = TX#tx.id,
+	case EmitMap of
+		#{ TXID := #{ peers := [] } } ->
+			{noreply, State};
+		#{ TXID := TXIDMap = #{ peers := [Peer | Peers] } } ->
+			spawn(
+				fun() ->
+					ar_http_iface_client:send_new_tx(Peer, TX),
+					erlang:send(?MODULE, {emitted_tx_to_peer, TX})
+				end
+			),
+			erlang:send(?MODULE, {emit_tx_to_peer, TX}),
+			{noreply, State#state{ emit_map = EmitMap#{ TXID => TXIDMap#{ peers => Peers } } }};
+		_ ->
+			{noreply, State}
+	end;
+
+handle_info({emitted_tx_to_peer, TX}, State = #state { emit_map = EmitMap }) ->
+	TXID = TX#tx.id,
+	case EmitMap of
+		#{ TXID := #{ peers := [], started_at := StartedAt } } ->
+			log_propagation_time(TX, timer:now_diff(erlang:timestamp(), StartedAt)),
+			erlang:send(?MODULE, {emitter_finished, TX}),
+			{noreply, State#state{ emit_map = maps:remove(TXID, EmitMap) }};
+		_ ->
+			{noreply, State}
+	end;
+
+handle_info({emitter_finished, TX}, State) ->
+	Bridge = whereis(http_bridge_node),
+	timer:apply_after(
+		ar_node_utils:calculate_delay(?TX_SIZE_BASE + byte_size(TX#tx.data)),
+		ar_bridge,
+		move_tx_to_mining_pool,
+		[Bridge, TX]
+	),
+	erlang:send_after(?EMITTER_INTER_WAIT, ?MODULE, emitter_go),
+	{noreply, State};
+
+handle_info(_Info, State) ->
+	{noreply, State}.
+
+terminate(_Reason, _State) ->
+	ok.
+
+code_change(_OldVsn, State, _Extra) ->
+	{ok, State}.
+
+format_status(_Opt, Status) ->
+	Status.
+
+%%%===================================================================
+%%% Private functions.
+%%%===================================================================
+
+maybe_drop(Q, Size, MaxSize) ->
+	maybe_drop(Q, Size, MaxSize, []).
+
+maybe_drop(Q, Size, MaxSize, DroppedTXs) ->
+	case Size > MaxSize of
+		true ->
+			{NewQ, TX, DroppedSize} = drop_lowest(Q),
+			maybe_drop(NewQ, Size - DroppedSize, MaxSize, [TX | DroppedTXs]);
+		false ->
+			{Q, Size, lists:filter(fun(TX) -> TX /= none end, DroppedTXs)}
+	end.
+
+%% @doc Drop lowest scoring item(s) from priority queue.
+%% NOTE relies on the internal implementation of priority_queue.
+drop_lowest(Q = {pqueue, []}) ->
```

Comment:
Since we need to look at this library's internals to be able to drop the lowest element, I wonder if we can use erlang's `gb_sets` with `{Priority, Item}` tuples to emulate a priority queue and not use this library at all.